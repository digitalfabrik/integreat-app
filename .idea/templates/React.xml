<templateSet group="React">
  <template name="rrc" value="import React, {ReactElement} from 'react';&#10;import { connect } from 'react-redux';&#10;&#10;type $TM_FILENAME_BASE$Props = {&#10;}&#10;type $TM_FILENAME_BASE$State = {&#10;}&#10;class $TM_FILENAME_BASE$ extends React.Component&lt;$TM_FILENAME_BASE$Props,$TM_FILENAME_BASE$State&gt; {&#10; constructor(props: $TM_FILENAME_BASE$Props) {&#10;  super(props);&#10;&#10;  this.state = {};&#10;&#10; }&#10;&#10; render() {&#10;  return (&#10;   &lt;div&gt;&#10;    $END$&#10;   &lt;/div&gt;&#10;  );&#10; }&#10;}&#10; &#10;const mapStateToProps: (state: IState) =&gt; MapStateToProps = state =&gt; ({&#10; &#10;});&#10;    &#10;export default connect(&#10;    mapStateToProps,&#10;)($componentName$);&#10;" description="Creates a React component class connected to redux" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="" defaultValue="" alwaysStopAt="true" />
    <variable name="componentName" expression="" defaultValue="&quot;MyComponent&quot;" alwaysStopAt="true" />
    <context>
      <option name="TypeScript" value="true" />
    </context>
  </template>
  <template name="rcfc" value="import React, { Component, ReactElement } from 'react';&#10;&#10;type $TM_FILENAME_BASE$Props = {&#10;}&#10;type $TM_FILENAME_BASE$State = {&#10;}&#10;&#10;class $componentName$ extends Component&lt;$TM_FILENAME_BASE$Props,$TM_FILENAME_BASE$State&gt; {&#10;    constructor(props: $TM_FILENAME_BASE$Props) {&#10;        super(props);&#10;    }&#10;    &#10;    componentWillMount() {&#10;    &#10;    }&#10;    &#10;    componentDidMount() {&#10;    &#10;    }&#10;    &#10;    componentWillReceiveProps(nextProps) {&#10;    &#10;    }&#10;    &#10;    shouldComponentUpdate(nextProps, nextState) {&#10;    &#10;    }&#10;    &#10;    componentWillUpdate(nextProps, nextState) {&#10;    &#10;    }&#10;    &#10;    componentDidUpdate(prevProps, prevState) {&#10;    &#10;    }&#10;    &#10;    componentWillUnmount() {&#10;    &#10;    }&#10;    &#10;    render(): ReactElement {&#10;        return (&#10;            &lt;div&gt;&#10;                $END$&#10;            &lt;/div&gt;&#10;        );&#10;    }&#10;}&#10;&#10;export default $componentName$;&#10;" description="Creates a React component class with all lifecycle methods and ES6 module system" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="" defaultValue="" alwaysStopAt="true" />
    <variable name="componentName" expression="" defaultValue="&quot;MyComponent&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="rfc" value="import React, {ReactElement} from 'react';&#10;&#10;type $TM_FILENAME_BASE$Props = {&#10;}&#10;&#10;const $TM_FILENAME_BASE$: React.FC&lt;$TM_FILENAME_BASE$Props&gt; = (props:$TM_FILENAME_BASE$Props):ReactElement =&gt; {&#10;    return (&#10;        &lt;div&gt;&#10;            $END$&#10;        &lt;/div&gt;&#10;    );&#10;};&#10;&#10;export default $TM_FILENAME_BASE$;&#10;" description="Creates a stateless React component with interface ES6 module system" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="con" value="constructor(props) {&#10;    super(props: $TM_FILENAME_BASE$Props);&#10;    $END$&#10;}&#10;" description="Adds a default constructor for the class that contains props as arguments" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="conc" value="constructor(props, context) {&#10;    super(props, context);&#10;    $END$&#10;}&#10;" description="Adds a default constructor for the class that contains props and context as arguments" toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="cwm" value="componentWillMount() {&#10;    $END$&#10;}&#10;" description="Invoked once, both on the client and server, immediately before the initial rendering occurs" toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="cdm" value="&#10;componentDidMount() {&#10;    $END$&#10;}&#10;" description="Invoked once, only on the client (not on the server), immediately after the initial rendering occurs." toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="scu" value="&#10;shouldComponentUpdate(nextProps, nextState) {&#10;    $END$&#10;}&#10;" description="Invoked before rendering when new props or state are being received." toReformat="true" toShortenFQNames="true">
    <context>
      <option name="JS_EXPRESSION" value="true" />
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="cwup" value="componentWillUpdate(nextProps, nextState) {&#10;    $END$&#10;}&#10;" description="Invoked immediately before rendering when new props or state are being received." toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="cdup" value="&#10;componentDidUpdate(prevProps, prevState) {&#10;    $END$&#10;}&#10;" description="Invoked immediately after the component's updates are flushed to the DOM." toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="cwun" value="componentWillUnmount() {&#10;    $END$&#10;}&#10;" description="Invoked immediately before a component is unmounted from the DOM." toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="ren" value="&#10;render(): ReactElement {&#10;    return (&#10;        &lt;div&gt;&#10;            $END$&#10;        &lt;/div&gt;&#10;    );&#10;}&#10;" description="When called, it should examine this.props and this.state and return a single child element." toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="ssf" value="&#10;this.setState((state, props) =&gt; { &#10;    return {&#10;        $END$ &#10;    };&#10;});&#10;" description="Performs a shallow merge of nextState into current state" toReformat="false" toShortenFQNames="true">
    <context />
  </template>
  <template name="rcc" value="import React, { Component } from 'react';&#10;&#10;class $TM_FILENAME_BASE$ extends Component {&#10; render() {&#10;  return (&#10;   &lt;div&gt;&#10;    $END$&#10;   &lt;/div&gt;&#10;  );&#10; }&#10;}&#10;&#10;export default $TM_FILENAME_BASE$;" description="Creates a A higher-order component" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="rpc" value="import React, {ReactElement} from 'react';&#10;&#10;type $TM_FILENAME_BASE$Props = {&#10;}&#10;type $TM_FILENAME_BASE$State = {&#10;}&#10;class $TM_FILENAME_BASE$ extends React.PureComponent&lt;$TM_FILENAME_BASE$Props,$TM_FILENAME_BASE$State&gt; {&#10; constructor(props: $TM_FILENAME_BASE$Props) {&#10;  super(props);&#10;&#10;  this.state = {};&#10;&#10; }&#10;&#10; render(): ReactElement {&#10;  return (&#10;   &lt;div&gt;&#10;    $END$&#10;   &lt;/div&gt;&#10;  );&#10; }&#10;}&#10;&#10;export default $TM_FILENAME_BASE$;" description="Creates a React pure component class with PropTypes and ES6 module system" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="rrdc" value="import React, { Component } from 'react';&#10;import { connect } from 'react-redux';&#10;&#10;type MapStateToProps = {&#10;}&#10;type MapDispatchToProps = {&#10;}&#10;&#10;type $TM_FILENAME_BASE$Props = MapStateToProps &amp; MapDispatchToProps &amp; {&#10;}&#10;type $TM_FILENAME_BASE$State = {&#10;}&#10;&#10;&#10;class $TM_FILENAME_BASE$ extends React.Component&lt;$TM_FILENAME_BASE$Props ,$TM_FILENAME_BASE$State&gt; {&#10; render() {&#10;  return (&#10;   &lt;div&gt;&#10;    $END$&#10;   &lt;/div&gt;&#10;  );&#10; }&#10;}&#10;&#10;const mapStateToProps: (state: IState) =&gt; MapStateToProps = state =&gt; ({&#10; &#10;});&#10;&#10;const mapDispatchToProps : (dispatch: Dispatch&lt;Action&gt;) =&gt; MapDispatchToProps = dispatch  =&gt; {&#10; return {&#10; }&#10;}&#10;&#10;export default connect(&#10; mapStateToProps, mapDispatchToProps&#10;)($TM_FILENAME_BASE$);" description="Creates a React component class connected to redux with dispatch" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="rccc" value="import React, {ReactElement} from 'react';&#10;&#10;type $TM_FILENAME_BASE$Props = {&#10;}&#10;type $TM_FILENAME_BASE$State = {&#10;}&#10;&#10;class $TM_FILENAME_BASE$ extends React.Component&lt;$TM_FILENAME_BASE$Props,$TM_FILENAME_BASE$State&gt; {&#10; constructor(props: $TM_FILENAME_BASE$Props) {&#10;  super(props);&#10;&#10;  this.state = {};&#10;&#10; }&#10;&#10; render(): ReactElement {&#10;  return (&#10;   &lt;div&gt;&#10;    $END$&#10;   &lt;/div&gt;&#10;  );&#10; }&#10;}&#10;&#10;export default $TM_FILENAME_BASE$;" description="Creates a React component class with constructor and interfaces" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="useState" value="const [$state$, set$setState$] = useState&lt;&gt;($initState$);$END$" description="useState" toReformat="true" toShortenFQNames="true">
    <variable name="state" expression="" defaultValue="&quot;state&quot;" alwaysStopAt="true" />
    <variable name="setState" expression="capitalize(state)" defaultValue="" alwaysStopAt="true" />
    <variable name="initState" expression="" defaultValue="&quot;initState&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="useEffect" value="useEffect(() =&gt; {&#10;$effect$&#10;  return () =&gt; {&#10;  // CleanUp&#10;    $effect$&#10;  };&#10;}, [$deps$]);&#10;$END$" description="useEffect" toReformat="true" toShortenFQNames="true">
    <variable name="effect" expression="" defaultValue="&quot;effect&quot;" alwaysStopAt="true" />
    <variable name="deps" expression="" defaultValue="&quot;deps&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="useCallback" value="const memoizedCallback = useCallback(&#10;  () =&gt; {&#10;    $callback$&#10;  },&#10;  [$input$],&#10;);&#10;$END$" description="useCallback" toReformat="true" toShortenFQNames="true">
    <variable name="callback" expression="" defaultValue="&quot;doSomething(a, b)&quot;" alwaysStopAt="true" />
    <variable name="input" expression="" defaultValue="&quot;a, b&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="useMemo" value="const memoizedValue = useMemo(() =&gt; computeExpensiveValue(a, b), [a, b]);$END$" description="useMemo" toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="rntc" value="import React from 'react'&#10;import { render } from '@testing-library/react-native'&#10;&#10;describe('$TM_FILENAME_BASE$', () =&gt; {&#10;  it('should render $TM_FILENAME_BASE$', () =&gt; {&#10;    const { getByText } = render(&lt;&gt;&lt;/&gt;)&#10;    expect(getByText('send')).toBeDisabled()&#10;  })&#10;})" description="Create standard component test with react-native testing-library" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rnfc" value="import React, {ReactElement} from 'react';&#10;import {View} from 'react-native'&#10;&#10;type $TM_FILENAME_BASE$Props = {&#10;}&#10;&#10;const $TM_FILENAME_BASE$: React.FC&lt;$TM_FILENAME_BASE$Props&gt; = (props:$TM_FILENAME_BASE$Props):ReactElement =&gt; {&#10;    return (&#10;        &lt;View&gt;&#10;            $END$&#10;        &lt;/View&gt;&#10;    );&#10;};&#10;&#10;export default $TM_FILENAME_BASE$;&#10;" description="Creates a stateless React-Native component with interface ES6 module system" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rnccc" value="import React, {ReactElement} from 'react';&#10;import {View} from 'react-native'&#10;&#10;type $TM_FILENAME_BASE$Props = {&#10;}&#10;type $TM_FILENAME_BASE$State = {&#10;}&#10;class $TM_FILENAME_BASE$ extends React.Component&lt;$TM_FILENAME_BASE$Props,$TM_FILENAME_BASE$State&gt; {&#10; constructor(props: $TM_FILENAME_BASE$Props) {&#10;  super(props);&#10;&#10;  this.state = {};&#10;&#10; }&#10;&#10; render(): ReactElement {&#10;  return (&#10;   &lt;View&gt;&#10;    $END$&#10;   &lt;/View&gt;&#10;  );&#10; }&#10;}&#10;&#10;export default $TM_FILENAME_BASE$;" description="Creates a React component class with constructor and interfaces" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="scnv" value="const $componentName$ = styled.View`&#10;`" description="styled view (native component)" toReformat="false" toShortenFQNames="true">
    <variable name="componentName" expression="" defaultValue="&quot;StyledView&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="scni" value="const $componentName$ = styled.Image`&#10;`" description="styled image(native component)" toReformat="true" toShortenFQNames="true">
    <variable name="componentName" expression="" defaultValue="&quot;StyledImage&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="scnt" value="const $componentName$ = styled.Text`&#10;`" description="styled text (native component)" toReformat="true" toShortenFQNames="true">
    <variable name="componentName" expression="" defaultValue="&quot;StyledText&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="useRef" value="const $ref$ = useRef&lt;&gt;($initialValue$);$END$" description="useRef" toReformat="true" toShortenFQNames="true">
    <variable name="ref" expression="" defaultValue="&quot;ref&quot;" alwaysStopAt="true" />
    <variable name="initialValue" expression="" defaultValue="&quot;initialValue&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="cref" value="this.$refName$Ref = React.createRef&lt;&gt;();$END$" description="Creates a ref" toReformat="true" toShortenFQNames="true">
    <variable name="refName" expression="" defaultValue="&quot;refName&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="hoc" value="import React from 'react'&#10;&#10;export default ($WrappedComponent$) =&gt; {&#10;  const $hocComponent$ = ({ ...props }) =&gt; &lt;$WrappedComponent$ {...props} /&gt;&#10;&#10;  return $hocComponent$&#10;}" description="Creates a React higher-order component" toReformat="true" toShortenFQNames="true">
    <variable name="WrappedComponent" expression="" defaultValue="&quot;WrappedComponent&quot;" alwaysStopAt="true" />
    <variable name="hocComponent" expression="" defaultValue="&quot;hocComponent&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="disp" value="const mapDispatchToProps : (dispatch: Dispatch&lt;Action&gt;) =&gt; MapDispatchToProps = dispatch  =&gt; {&#10; return {&#10; }&#10;}" description="Adds the redux mapDispatchToProps function" toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="rnpc" value="import React, {ReactElement} from 'react';&#10;import {View} from 'react-native'&#10;&#10;type $TM_FILENAME_BASE$Props = {&#10;}&#10;type $TM_FILENAME_BASE$State = {&#10;}&#10;&#10;class $TM_FILENAME_BASE$ extends React.PureComponent&lt;$TM_FILENAME_BASE$Props,$TM_FILENAME_BASE$State&gt; {&#10; constructor(props: $TM_FILENAME_BASE$Props) {&#10;  super(props);&#10;&#10;  this.state = {};&#10;&#10; }&#10;&#10; render(): ReactElement {&#10;  return (&#10;   &lt;View&gt;&#10;    $END$&#10;   &lt;/View&gt;&#10;  );&#10; }&#10;}&#10;&#10;export default $TM_FILENAME_BASE$;" description="Creates a React pure component class with PropTypes and ES6 module system" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rtc" value="import React from 'react'&#10;import { render } from '@testing-library/react'&#10;&#10;describe('$TM_FILENAME_BASE$', () =&gt; {&#10;  it('should render $TM_FILENAME_BASE$', () =&gt; {&#10;    const { getByText } = render(&lt;&gt;&lt;/&gt;)&#10;    expect(getByText('send')).toBeDisabled()&#10;  })&#10;})" description="Create standard component test with react testing-library" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
</templateSet>