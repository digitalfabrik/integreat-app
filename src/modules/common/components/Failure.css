.center-text {
  display: block;
  margin-top: 50px;
  text-align: center;
}
