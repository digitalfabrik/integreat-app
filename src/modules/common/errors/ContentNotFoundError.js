// @flow

import { ResponseError } from '@integreat-app/integreat-api-client'

type NotFoundType = 'category' | 'event' | 'extra' | 'poi' | 'newsItem' | 'tuNewsItem'

class ContentNotFoundError extends Error {
  _type: NotFoundType
  _id: string | number
  _city: string
  _language: string

  getMessage = (type: NotFoundType, id: string | number): string =>
    `The ${type} ${id} does not exist here.`

  constructor (params: { type: NotFoundType, id: string | number, city: string, language: string }) {
    super()
    this.message = this.getMessage(params.type, params.id)
    this._type = params.type
    this._id = params.id
    this._city = params.city
    this._language = params.language

    // https://github.com/babel/babel/issues/3083
    /* eslint-disable */
    // $FlowFixMe
    this.constructor = ContentNotFoundError
    // $FlowFixMe
    this.__proto__ = ContentNotFoundError.prototype
    /* eslint-enable */

    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, ResponseError)
    }
  }

  get type (): NotFoundType {
    return this._type
  }

  get id (): string | number {
    return this._id
  }

  get city (): string {
    return this._city
  }

  get language (): string {
    return this._language
  }
}

export default ContentNotFoundError
