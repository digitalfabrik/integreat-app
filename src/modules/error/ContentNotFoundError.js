// @flow

type NotFoundType = 'category' | 'event' | 'extra'

const getMessage = (type: NotFoundType, id: string): string =>
  `The ${type} ${id} does not exist here.`

class ContentNotFoundError extends Error {
  _type: NotFoundType
  _id: string | number
  _city: string
  _language: string

  constructor (params: { type: NotFoundType, id: string, city: string, language: string }) {
    super(getMessage(params.type, params.id))

    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, ContentNotFoundError)
    }

    this.name = 'ContentNotFoundError'
    this._type = params.type
    this._id = params.id
    this._city = params.city
    this._language = params.language
  }

  get type (): NotFoundType {
    return this._type
  }

  get id (): string | number {
    return this._id
  }

  get city (): string {
    return this._city
  }

  get language (): string {
    return this._language
  }
}

export default ContentNotFoundError
