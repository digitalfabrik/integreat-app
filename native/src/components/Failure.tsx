import React, { ReactElement } from 'react'
import styled from 'styled-components/native'
import { Text } from 'react-native'
import { useTranslation } from 'react-i18next'
import FailureIcon from '../assets/FailureIcon.svg'
import { Button } from 'react-native-elements'
import { ErrorCode } from 'api-client'
import { ThemeType } from 'build-configs'

const ViewContainer = styled.View`
  flex: 1;
  align-items: center;
  margin-top: 15%;
`
const IconContainer = styled.Image`
  margin-bottom: 10px;
`

export type PropsType = {
  code: ErrorCode
  tryAgain?: () => void
  theme: ThemeType
}

const Failure = ({ code, tryAgain, theme }: PropsType): ReactElement => {
  const { t } = useTranslation('error')
  return (
    <ViewContainer>
      <IconContainer source={FailureIcon} />
      <Text>{t(code)}</Text>
      {tryAgain && (
        <Button
          testID='button-tryAgain'
          titleStyle={{
            color: theme.colors.textColor
          }}
          buttonStyle={{
            backgroundColor: theme.colors.themeColor,
            marginTop: 20
          }}
          onPress={tryAgain}
          title={t('tryAgain')}
        />
      )}
    </ViewContainer>
  )
}

export default Failure
