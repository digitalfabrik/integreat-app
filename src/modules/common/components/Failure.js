import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'react-i18next'
import FontAwesome from 'react-fontawesome'

import style from './Failure.css'
import { Link } from 'redux-little-router'

/**
 * Our error component, but since the name Error collides with the ES6 class, we've called it Failure
 */
export class Failure extends React.Component {
  static propTypes = {
    error: PropTypes.string.isRequired,
    t: PropTypes.func.isRequired
  }

  render () {
    const {t, error} = this.props
    return <div>
      <div className={style.centerText}>{t(error)}</div>
      <div className={style.centerText}><FontAwesome name='frown-o' size='5x' /></div>
      <Link className={style.centerText} href={'/'}>{t('goToStart')}</Link>
    </div>
  }
}

export default translate('common')(Failure)
