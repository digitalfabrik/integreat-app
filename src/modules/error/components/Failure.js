// @flow

import React from 'react'
import styled from 'styled-components/native'
import { type StyledComponent } from 'styled-components'
import { Text } from 'react-native'
import type { TFunction } from 'react-i18next'
import FailureIcon from '../assets/FailureIcon.svg'
import type { ThemeType } from '../../theme/constants/theme'
import { Button } from 'react-native-elements'
import type { ErrorCodeType } from '../ErrorCodes'

const ViewContainer: StyledComponent<{}, ThemeType, *> = styled.View`
flex: 1;
align-items: center;
margin-top: 15%;
`

const IconContainer = styled.Image`
margin-bottom: 10px;
`

type PropsType = {|
  errorMessage?: string,
  code: ErrorCodeType,
  tryAgain?: () => void,
  t: TFunction,
  theme: ThemeType
|}

class Failure extends React.Component<PropsType> {
  render () {
    const { t, code, tryAgain, theme } = this.props

    return <ViewContainer>
      <IconContainer source={FailureIcon} />
      <Text>{t(code)}</Text>
      {tryAgain &&
      <Button testID='button-tryAgain' titleStyle={{ color: theme.colors.textColor }}
              buttonStyle={{ backgroundColor: theme.colors.themeColor, marginTop: 20 }}
              onPress={tryAgain} title={t('tryAgain')} />}
    </ViewContainer>
  }
}

export default Failure
