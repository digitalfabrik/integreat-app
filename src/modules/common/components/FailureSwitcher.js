// @flow

import * as React from 'react'
import ContentNotFoundError from '../errors/ContentNotFoundError'
import Failure from './Failure'
import type { TFunction } from 'react-i18next'
import { withTranslation } from 'react-i18next'
import CategoriesRouteConfig from '../../app/route-configs/CategoriesRouteConfig'
import EventsRouteConfig from '../../app/route-configs/EventsRouteConfig'
import NewsRouteConfig from '../../app/route-configs/NewsRouteConfig'
import ExtrasRouteConfig from '../../app/route-configs/ExtrasRouteConfig'
import PoisRouteConfig from '../../app/route-configs/PoisRouteConfig'
import LocalNewsDetailsRouteConfig from '../../app/route-configs/LocalNewsDetailsRouteConfig'

import TuNewsListRouteConfig from '../../app/route-configs/TuNewsListRouteConfig'

type PropsType = {|
  error: Error,
  t: TFunction
|}

export class FailureSwitcher extends React.Component<PropsType> {
  /**
   * Renders a Failure with a link to the "home" of the route and information about what was not found
   * @param error
   * @return {*}
   */
  static renderContentNotFoundComponent (error: ContentNotFoundError): React.Node {
    const { city, language } = error
    switch (error.type) {
      case 'category':
        return <Failure goToPath={new CategoriesRouteConfig().getRoutePath({ city, language })}
                        goToMessage='goTo.categories'
                        errorMessage='notFound.category' />
      case 'event':
        return <Failure goToPath={new EventsRouteConfig().getRoutePath({ city, language })}
                        goToMessage='goTo.events'
                        errorMessage='notFound.event' />
      case 'newsItem':
        return <Failure goToPath={new LocalNewsDetailsRouteConfig().getRoutePath({ city, language })}
                        goToMessage='goTo.news'
                        errorMessage='notFound.newsItem' />
      case 'tuNewsItem':
        return <Failure goToPath={new TuNewsListRouteConfig().getRoutePath({ city, language })}
                        goToMessage='goTo.tuNews'
                        errorMessage='notFound.tuNewsItem' />

      case 'extra':
        return <Failure goToPath={new ExtrasRouteConfig().getRoutePath({ city, language })}
                        goToMessage='goTo.extras'
                        errorMessage='notFound.extra' />
      case 'poi':
        return <Failure goToPath={new PoisRouteConfig().getRoutePath({ city, language })}
                        goToMessage='goTo.pois'
                        errorMessage='notFound.poi' />
    }

    throw new Error('Failed to find component to render a content error')
  }

  render () {
    const error = this.props.error
    if (error instanceof ContentNotFoundError) {
      return FailureSwitcher.renderContentNotFoundComponent(error)
    } else {
      return <Failure errorMessage={error.message} />
    }
  }
}

export default withTranslation('error')(FailureSwitcher)
