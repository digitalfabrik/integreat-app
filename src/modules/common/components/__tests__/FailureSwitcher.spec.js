// @flow

import React from 'react'
import { mount } from 'enzyme'
import { FailureSwitcher } from '../FailureSwitcher'
import ContentNotFoundError from '../../errors/ContentNotFoundError'
import { Failure } from '../Failure'

describe('FailureSwitcher', () => {
  const city = 'augsburg'

  const language = 'de'

  describe.each`
    type                | id
    ${'category'}       | ${'willkommen'}
    ${'event'}          | ${'1234'}
    ${'localNewsItem'}  | ${'/augsburg/en/news/local/1'}
    ${'tunewsItem'}     | ${'/augsburg/en/news/tu-news/1'}
    ${'extra'}          | ${'sprungbrett'}
    ${'poi'}            | ${'1234'}
    `('render $type component not found failure',
  ({ type, id }) => {
    it(`should render a ${type} not found failure and match snapshot`, () => {
      const error = new ContentNotFoundError({ type, id, language, city })
      const component = FailureSwitcher.renderContentNotFoundComponent(error)
      expect(component).toMatchSnapshot()
    })

    it('should call render content not found component and create a Failure component', () => {
      const error = new ContentNotFoundError({ type, id, language, city })
      const spy = jest.spyOn(FailureSwitcher, 'renderContentNotFoundComponent')
      const component = mount(<FailureSwitcher error={error} />)
      expect(spy).toHaveBeenCalledWith(error)
      expect(component.find(Failure)).toHaveLength(1)

      spy.mockRestore()
    })
  })

  it('should render a failure as default', () => {
    const error = new Error('error message')
    const component = mount(<FailureSwitcher error={new Error('error message')} />)

    expect(component.find(Failure)).toHaveLength(1)
    expect(component.find(Failure).at(0).props().errorMessage).toEqual(error.message)
  })
})
