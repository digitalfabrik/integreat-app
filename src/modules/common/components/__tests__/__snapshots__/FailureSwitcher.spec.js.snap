// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FailureSwitcher render component not found failure should render a category not found failure and match snapshot 1`] = `
<Translate(Failure)
  errorMessage="not-found.category"
  goToAction={
    Object {
      "payload": Object {
        "categoryPath": undefined,
        "city": "augsburg",
        "language": "de",
      },
      "type": "CATEGORIES",
    }
  }
  goToMessage="goTo.categories"
/>
`;

exports[`FailureSwitcher render component not found failure should render a event not found failure and match snapshot 1`] = `
<Translate(Failure)
  errorMessage="not-found.event"
  goToAction={
    Object {
      "payload": Object {
        "city": "augsburg",
        "eventId": undefined,
        "language": "de",
      },
      "type": "EVENTS",
    }
  }
  goToMessage="goTo.events"
/>
`;

exports[`FailureSwitcher render component not found failure should render a extra not found failure and match snapshot 1`] = `
<Translate(Failure)
  errorMessage="not-found.extra"
  goTo={
    Object {
      "payload": Object {
        "city": "augsburg",
        "language": "de",
      },
      "type": "EXTRAS",
    }
  }
  goToMessage="goTo.extras"
/>
`;

exports[`FailureSwitcher should render a LanguageFailure if there is a LanguageNotFoundError 1`] = `
<React.Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      not-found.pageTitle
    </title>
  </HelmetWrapper>
  <Connect(Translate(LanguageFailure))
    city="augsburg"
  />
</React.Fragment>
`;

exports[`FailureSwitcher should render a content not found failure if there is a ContentNotFoundError 1`] = `
<React.Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      not-found.pageTitle
    </title>
  </HelmetWrapper>
  <Translate(Failure)
    errorMessage="not-found.extra"
    goTo={
      Object {
        "payload": Object {
          "city": "augsburg",
          "language": "de",
        },
        "type": "EXTRAS",
      }
    }
    goToMessage="goTo.extras"
  />
</React.Fragment>
`;

exports[`FailureSwitcher should render a failure as default 1`] = `
<React.Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      pageTitle
    </title>
  </HelmetWrapper>
  <Translate(Failure)
    errorMessage="error message"
  />
</React.Fragment>
`;
